src/tcgwars/logic/impl/gen7/CosmicEclipse.groovy

```diff
@@ -1498,9 +1505,12 @@ public enum CosmicEclipse implements LogicCardInfo {
           move "Direct Dive", {
             text "Discard all Energy from this Pokémon. This attack does 100 damage to 1 of your opponent's Benched Pokémon. (Don't apply Weakness and Resistance for Benched Pokémon.)"
             energyCost W, W, W
+            attackRequirement {
+              assert opp.bench : "Your opponent has no Benched Pokémon."
+            }
```

Comment:
Might require an actual official ruling (since the only one similar to this is from 12 years ago). But I'd say you can use the attack, discard the Pokémon Energy and not attack due to lack of bench. The discard to me seems like an effect, not a cost. Other cards should be updated based on how this is handled:

* Talonflame (UNM)
* Kingdra (BUS)
* Arceus (DP Promo DP50)
* Hippowdon LV.X (Rising Rivals 107)
* Blastoise (Platinum 2)
* Ampharos (POP7 1, Secret Wonders 1)

From Compendium LV.X:

> Q. Can you use Salamence-EX's "Hydro Wave" attack if the opponent has no Benched Pokemon?
>
> A. Yes. Salamence-EX would simply discard all attached Water Energy and that's all. (Apr 3, 2008 PUI Rules Team)
>
> ![](https://i.imgur.com/8OtJkK3.png)

Reply:
You're probably right. It still has an effect even if it does no damage. I'll correct this.